main.go

```diff
@@ -231,6 +233,7 @@ func main() { // nolint:gocyclo
 			&corev1.PersistentVolumeClaim{}: cacheLabelConfig,
 			&corev1.ConfigMap{}:             cacheLabelConfig, // Matching just labeled ConfigMaps and Secrets greatly reduces cache size
 			&corev1.Secret{}:                cacheLabelConfig, // Omitting labels or supporting custom labels would require changes in Grafana Reconciler
+			&gwapiv1.HTTPRoute{}:            cacheLabelConfig,
```

Comment:
As it relies on a CRD that might not be present, I think the correct implementation would be to have a discovery check like for OpenShift Routes (check how `isOpenShift` is calculated).
With Routes, we've seen the operator panicking in non-OpenShift environments, that's why Route caching is hidden behind the `isOpenShift` check.